addons/mrp/models/stock_rule.py

```diff
@@ -110,8 +110,10 @@ def _run_manufacture(self, procurements):
 
         for company_id in new_productions_values_by_company:
             productions_vals_list = new_productions_values_by_company[company_id]['values']
-            # create the MO as SUPERUSER because the current user may not have the rights to do it (mto product launched by a sale for example)
-            productions = self.env['mrp.production'].with_user(SUPERUSER_ID).sudo().with_company(company_id).create(productions_vals_list)
+            # Create the MO using the current logged-in user if the replenishment is triggered manually.
+            # Otherwise, fallback to SUPERUSER_ID (e.g., MO generated from sales orders).
+            user_id = (self.env.context.get('manual_replenishment') and self.env.uid) or SUPERUSER_ID
```

Comment:
```suggestion
            ctx = self.env.context
            user_id = self.env.uid if (ctx.get("manual_replenishment") or ctx.get("from_orderpoint")) and not ctx.get("ir_cron_progress_id") else SUPERUSER_ID
```
Can't we do something like this? Here, if the schedular is manually called by current user then still this `_run_buy` and `_run_manufacuturre` should be called, at that time the current user is not tracked.